core/src/main/java/com/linecorp/armeria/client/HttpResponseDecoder.java

```diff
@@ -286,8 +286,15 @@ private void close(@Nullable Throwable cause,
             if (cancelTimeout()) {
                 actionOnTimeoutCancelled.accept(cause);
             } else {
-                if (cause != null && !Exceptions.isExpected(cause)) {
-                    logger.warn("Unexpected exception:", cause);
+                if (cause != null && logger.isWarnEnabled() && !Exceptions.isExpected(cause)) {
+                    StringBuilder logMsg = new StringBuilder("Unexpected exception while closing");
+                    if (request != null) {
```

Comment:
I think we need to add:
`request.authority() != null`

Reply:
oh, is there a risk that `request.authority()` is null?

Reply:
I don't think so. Authority is nullable only for user convenience when sending a client request (so that Armeria auto-fills it), i.e. it is non-null on the server side. To make IntelliJ IDEA inspection happy, you might want to add an assertion though.

Reply:
Never mind. This is client-side and thus authority can be null. Sorry! :sweat_smile: